controller/state/statedb.go

```diff
@@ -328,20 +360,66 @@ func (s *stateDB) AssignTask(ctx context.Context, req tasks.PopTask) (tasks.Task
 		}
 	}
 
+	// Keep popping tasks until there a runanable task is found or until there
+	// are no more tasks
 	var assigned tasks.Task
+	for {
+		task, scheduled, err := s.popTask(ctx, req.WorkedBy.String(), &req.Status, tags)
+		if err != nil {
+			return nil, err
+		}
+		if task == nil {
+			// No more tasks to pop
+			return nil, nil
+		}
+		if !scheduled {
+			// Found a task that is runable, stop looking for tasks
+			assigned = task
+			break
+		}
+		// Got a scheduled task, so schedule it.
+		err = s.scheduleTask(task, nil)
```

Comment:
This seems like a really unusual time to schedule a task -- it requires a daemon to attempt to pop it? What if daemons are busy and you miss the first schedule? Why not put scheduling when you initially create the task?

Reply:
Was thinking, why schedule tasks if no one is asking for them.  If we want to start up a controller and load a bunch of tasks, but then sometime later start daemons, this may be nice.  I suppose that whether it is better to worse to miss the first schedule depends on the type of schedule (run every n hours, or run at a specific time).

So, will move scheduling to point of ingestion as that is probably least surprising.

Reply:
Tasks are now scheduled on ingestion.  However, I left the ability to detect and schedule unassigned scheduled tasks here so that if we want to insert a scheduled task into the DB outside of the normal ingestion path, it will still get picked up.  Otherwise, this the code here should never see a scheduled task.

Reply:
I would say we specifically want things coming in through the normal ingestion path -- NewStorageTask and NewRetrievalTask, and nothing else. I'm trying to think what else could happen here -- inserting through PG Admin? I'd hope we never do that.

But I don't feel so strongly I'd block the PR on it.

Reply:
If `popTask` finds an unassigned task that has a schedule, there are two things that can be done:
1. Schedule the task (what is done now)
2. Log error and ignore task

With the way it is now (option 1), the new dealbot will pick up any unassigned scheduled tasks that the previous dealbot may not have popped yet.  With option 2, any of those tasks would have to be removed and re-inserted.

Reply:
Logging a warning when a scheduled task with no owner is seen.